Library/ViewModels/PledgeViewModel.swift

```diff
@@ -860,38 +860,75 @@ public class PledgeViewModel: PledgeViewModelType, PledgeViewModelInputs, Pledge
         )
       }
 
-    Signal
-      .combineLatest(
-        createBackingData,
-        baseReward,
-        additionalPledgeAmount,
-        allRewardsShippingTotal
+    // Emmits value when Pledge Button is clicked.
+
+    Signal.combineLatest(
+      createBackingData,
+      baseReward,
+      additionalPledgeAmount,
+      allRewardsShippingTotal
+    )
+    .takeWhen(createButtonTapped)
+    .map { data, baseReward, additionalPledgeAmount, allRewardsShippingTotal in
```

Comment:
Is there a way to re-factor the map so we aren't repeating it word-for-word twice here as well?

Reply:
Not really, because a property on the `isApplePay` field is different in both maps, if we had Lenses for CheckoutPropertiesData, maybe we can manipulate things a bit.

Reply:
I did a bit of refactor to CheckoutPropertiesData in the map so it can be reused, but in my opinion it's not necessary since we only need it in those two places.